lib/pacemaker/pcmk_sched_group.c

```diff
@@ -356,19 +356,18 @@ group_internal_constraints(pe_resource_t * rsc, pe_working_set_t * data_set)
 }
 
 void
-group_rsc_colocation_lh(pe_resource_t *dependent, pe_resource_t *primary,
-                        pcmk__colocation_t *constraint)
+pcmk__group_apply_coloc_score(pe_resource_t *dependent, pe_resource_t *primary,
+                              pcmk__colocation_t *constraint,
+                              bool for_dependent)
 {
     GList *gIter = NULL;
     group_variant_data_t *group_data = NULL;
 
-    if (dependent == NULL) {
-        pe_err("dependent was NULL for %s", constraint->id);
-        return;
+    CRM_CHECK((constraint != NULL) && (dependent != NULL) && (primary != NULL),
+              return);
 
-    } else if (primary == NULL) {
-        pe_err("primary was NULL for %s", constraint->id);
-        return;
+    if (!for_dependent) {
+        goto for_primary;
```

Comment:
We could make `pcmk__group_apply_coloc_score()` a wrapper for two static functions (one for dependent, one for primary), since both cases have non-trivial logic. We could have shorter functions and skip the goto jump, and everyone else can still call `pcmk__group_apply_coloc_score()` instead of dealing with the LH/RH mess again. Thoughts?

Reply:
Yep, I'll take care of that when I get to groups